warehouse/query-core/src/main/java/datawave/query/jexl/JexlNodeFactory.java

```
@@ -108,142 +112,151 @@ public static JexlNode createNodeTreeFromFieldsToValues(ContainerType containerT
+            for (String identifier : JexlASTHelper.getIdentifierNames(original)) {
                 fieldsToValues.put(identifier, allValues);
             }
         }
         
-        Set<String> fields = fieldsToValues.keySet();
+        List<JexlNode> children = new LinkedList<>();
         
-        JexlNode parentNode = (containerType.equals(ContainerType.OR_NODE) ? new ASTOrNode(ParserTreeConstants.JJTORNODE) : new ASTAndNode(
-                        ParserTreeConstants.JJTANDNODE));
-        int parentNodeChildCount = 0;
+        Set<String> fields = fieldsToValues.keySet();
         
         if (keepOriginalNode) {
-            JexlNodes.ensureCapacity(parentNode, fields.size() + 1);
-            JexlNode child = RebuildingVisitor.copy(orgNode);
-            parentNode.jjtAddChild(child, parentNodeChildCount);
-            child.jjtSetParent(parentNode);
-            parentNodeChildCount++;
+            JexlNode child = copy(original);
+            children.add(child);
             // remove this entry from the fieldsToValues to avoid duplication
-            for (String identifier : JexlASTHelper.getIdentifierNames(orgNode)) {
-                for (Object value : JexlASTHelper.getLiteralValues(orgNode)) {
+            for (String identifier : JexlASTHelper.getIdentifierNames(original)) {
+                for (Object value : JexlASTHelper.getLiteralValues(original)) {
                     fieldsToValues.remove(identifier, value);
                 }
             }
-        } else {
-            JexlNodes.ensureCapacity(parentNode, fields.size());
         }
         
         for (String field : fields) {
-            ValueSet valuesForField = fieldsToValues.get(field);
+            JexlNode child = createNodeFromValuesForField(field, fieldsToValues, original, containerType, expandValues, isNegated);
+            if (child != null) {
+                JexlASTHelper.validateLineage(child, true);
```

this is the only place I see we are failing hard in non-test code

---

I've removed this in the next commit, I was building and looking for failure cases.